extensions/aerosol_optics/mo_aerosol_optics.F90

```diff
@@ -0,0 +1,633 @@
+! Regents of the University of Colorado.  All right reserved.
+!
+! Use and duplication is permitted under the terms of the
+!    BSD 3-clause license, see http://opensource.org/licenses/BSD-3-Clause
+! -------------------------------------------------------------------------------------------------
+! Provides aerosol optical properties as a function of effective radius for the RRTMGP bands
+!   Based on climatoligical aerosol optical properties used in MERRA2 as derived from
+!     the GOCART model for 15 aerosol types, including five dust types, five sea salt types,
+!     one sulfate type, and both hydrophobic and hydrophilic black carbon and organic carbon.
+!   Input aerosol optical data are stored in look-up tables.
+!
+!   References for gocart interactive aerosols:                        
+!     Chin et al., 2000 - jgr, v105, 24671-24687                       
+!     Colarco et al., 2010 - jgr, v115, D14207                         
+!                                                                      
+!   References for merra2 aerosol reanalysis:                          
+!     Randles et al., 2017 - jclim, v30, 6823-6850                     
+!     Buchard et al., 2017 - jclim, v30, 6851-6871                     
+!                                                                      
+! The class can be used as-is but is also intended as an example of how to extend the RTE framework
+! -------------------------------------------------------------------------------------------------
+
+module mo_aerosol_optics
+  use mo_rte_kind,      only: wp, wl
+  use mo_rte_config,    only: check_values, check_extents
+  use mo_rte_util_array,only: any_vals_less_than, any_vals_outside, extents_are
+  use mo_optical_props, only: ty_optical_props,      &
+                              ty_optical_props_arry, &
+                              ty_optical_props_1scl, &
+                              ty_optical_props_2str, &
+                              ty_optical_props_nstr
+  implicit none
+
+  ! MERRA2/GOCART aerosol types
+  integer, parameter, public :: merra_aero_dust = 1      ! Dust
+  integer, parameter, public :: merra_aero_salt = 2      ! Salt
+  integer, parameter, public :: merra_aero_sulf = 3      ! sulfate
+  integer, parameter, public :: merra_aero_bcar_rh = 4   ! black carbon, hydrophilic
+  integer, parameter, public :: merra_aero_bcar = 5      ! black carbon, hydrophobic
+  integer, parameter, public :: merra_aero_ocar_rh = 6   ! organic carbon, hydrophilic
+  integer, parameter, public :: merra_aero_ocar = 7      ! organic carbon, hydrophobic
+
+  private
+  ! -----------------------------------------------------------------------------------
+  type, extends(ty_optical_props), public :: ty_aerosol_optics
+    private
+    !
+    ! Lookup table information
+    !
+    ! Table upper and lower size bin limits and effective (center) values
+    real(wp),dimension(:), allocatable :: aero_bin_hi(:)
```

Comment:
@mjiacono The `ty_gas_optics` class of which this class is a child has a [way of describing the spectral discretization](https://github.com/earth-system-radiation/rte-rrtmgp/blob/83ff9b315953506333c9143c0b6538d8f7020244/rte/mo_optical_props.F90#L79). Can we adopt the same conventions, i.e. a single vector describing the bounds of the bins? 

Please follow [RTE conventions](https://earth-system-radiation.github.io/rte-rrtmgp/reference/index.html) and use MKS (meters/kilograms/seconds)

Reply:
@RobertPiincus Replaced aero_bin_lo(nbin) and aero_bin_hi(nbin) with merra_aero_bin_lims(npair,nbin) and removed the former arrays and aero_bin_eff (this is the effective center of each size bin, which was provided by MERRA, but we won’t need).